service/src/main/scala/com/yahoo/maha/service/MahaService.scala

```diff
@@ -657,4 +662,139 @@ object MahaServiceConfig {
+      postRegistryContext = postExecutorContext.copy(registryMap = registryMap)
+      parallelServiceExecutorConfig <- initParallelServiceExecutors(jsonMahaServiceConfig.parallelServiceExecutorConfigMap)(postRegistryContext)
+      postParallelServiceExecutorContext = postRegistryContext.copy(parallelServiceExecutorMap = parallelServiceExecutorConfig)
+      curatorMap <- initCurators(jsonMahaServiceConfig.curatorMap)(postParallelServiceExecutorContext)
+      postCuratorContext = postParallelServiceExecutorContext.copy(curatorMap = curatorMap)
+      mahaRequestLogWriter <- initKafkaLogWriter(jsonMahaServiceConfig.jsonMahaRequestLogConfig)(postCuratorContext)
+    } yield {
+      val objectNameMap = mergeMaps(bucketConfigMap, utcTimeProviderMap, generatorMap, executorMap, registryMap, parallelServiceExecutorConfig, curatorMap)
+      val dynamicProperties = findDynamicProperties(json, objectNameMap.toMap)
+      for ((_, dynamicProperty) <- dynamicProperties) {
+        dynamicProperty.objects.keys.foreach(objName => {
+          val dynamicObject = getDynamicObject(objectNameMap(objName))
+          objectNameMap.put(objName, dynamicObject)
+        })
+      }
+
+      val resultMap: Map[String, RegistryConfig] = registryMap.map {
+        case (regName, registry) => {
+          val registryConfig = jsonMahaServiceConfig.registryMap.get(regName.toLowerCase).get
+
+          implicit val queryGeneratorRegistry = new QueryGeneratorRegistry
+          generatorMap.filter(g => registryConfig.generators.contains(g._1)).foreach {
+            case (name, generator) =>
+              queryGeneratorRegistry.register(generator.engine, generator)
+          }
+          val queryExecutorContext = new QueryExecutorContext
+          executorMap.filter(e => registryConfig.executors.contains(e._1)).foreach {
+            case (_, executor) =>
+              queryExecutorContext.register(executor)
+          }
+          (regName -> RegistryConfig(regName,
+            registry,
+            new DefaultQueryPipelineFactory(),
+            queryExecutorContext,
+            new BucketSelector(registry, bucketConfigMap.get(registryConfig.bucketConfigName).get),
+            utcTimeProviderMap.get(registryConfig.utcTimeProviderName).get,
+            parallelServiceExecutorConfig.get(registryConfig.parallelServiceExecutorName).get))
+        }
+      }
+      DynamicMahaServiceConfig(dynamicProperties, postCuratorContext, resultMap, mahaRequestLogWriter, curatorMap)
+    }
+    dynamicMahaServiceConfig
+  }
+
+  def mergeMaps(maps: Map[String, Object]*): mutable.Map[String, Object] = {
+    val mergedMap = new scala.collection.mutable.HashMap[String, Object]()
+    maps.foreach(map => {
+      map.foreach(f => mergedMap.put(f._1, f._2))
+    })
+    mergedMap
+  }
+
+  def findDynamicProperties(json: JValue, objectNameMap: Map[String, Object]): Map[String, DynamicPropertyInfo] = {
```

Comment:
objectNameMap has String-> Object, we should use the base trait(can be name as MscObject)  instead of generic object to enforce the implementation in registry configs.

Reply:
Can't do that right now because ParallelServiceExecutor is in par-request-2 module which doesn't depend on any other maha module (like core). Will need some refactoring to do it cleanly.